Update how-to-handle-multiple-events-using-event-properties.md (#23414)
dotnet/docs: docs/standard/events/how-to-handle-multiple-events-using-event-properties.md

---
# How to: Handle Multiple Events Using Event Properties

> To use event properties, you define the event properties in the class that raises the events, and then set the delegates for the event properties in classes that handle the events. To implement multiple event properties in a class, the class should internally store and maintain the delegate defined for each event. A typical approach is to implement a delegate collection that is indexed by an event key. Because each field-like-event requires a reference backing-field. And when the number of events increases, a lot of fields will be created even they are not subscribed. This can introduce unnecessary allocations when only a few events are subscribed.

Reviewer: ```suggestion
To use event properties, you define the event properties in the class that raises the events, and then set the delegates for the event properties in classes that handle the events. To implement multiple event properties in a class, the class should internally store and maintain the delegate defined for each event.

As the number of events increases, so too does the number of backing-field references. To help avoid unnecessary allocations an approach is to implement a delegate collection that is indexed by an event key.
```

Author: "As the number of events increases" is imprecise, as you can see from the code example, even with self-defined add/remove accessors the number of events still increases anyway. The issue only happens when you rely on the complier's syntax sugar, a "field-like-event", which will generate a backing-field:
`public event EventHandler<EventArgs> MyEvent;`

Reviewer: I updated my suggestion, please have another look - and let me know what you think.

Author: HI, IMO it's generally good😊, but why "must" again? As you can see in you example `Progress`  and `AnotherProgress` generated backing-field, they still work but suffer from the issue when the number of events increases, but `ManualProgress` doesn't.

Author: I've updated the doc according to your suggestion. Take a look
